pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -1936,18 +1937,48 @@ func (a *ApplicationBackupController) backupResources(
 		}
 
 		if len(resourceTypeNsBatch) != 0 {
-			for _, backupResourceType := range backup.Spec.ResourceTypes {
-				for _, resource := range resourceTypes {
-					if resource.Kind == backupResourceType || (backupResourceType == "PersistentVolumeClaim" && resource.Kind == "PersistentVolume") {
-						log.ApplicationBackupLog(backup).Tracef("GetResourcesType for : %v", resource.Kind)
-						objects, _, err := a.resourceCollector.GetResourcesForType(resource, nil, resourceTypeNsBatch, backup.Spec.Selectors, nil, nil, true, resourceCollectorOpts)
-						if err != nil {
-							log.ApplicationBackupLog(backup).Errorf("Error getting resources: %v", err)
-							return err
+			if len(backup.Spec.ResourceTypes) != 0 {
+				for _, backupResourceType := range backup.Spec.ResourceTypes {
+					for _, resource := range resourceTypes {
+						if resource.Kind == backupResourceType ||
+							(backupResourceType == "PersistentVolumeClaim" && resource.Kind == "PersistentVolume") {
+							log.ApplicationBackupLog(backup).Tracef("GetResourcesType for : %v", resource.Kind)
+							objects, _, err := a.resourceCollector.GetResourcesForType(resource, nil, resourceTypeNsBatch,
+								backup.Spec.Selectors, nil, nil, true, resourceCollectorOpts)
+							if err != nil {
+								log.ApplicationBackupLog(backup).Errorf("Error getting resources: %v", err)
+								return err
+							}
+							allObjects = append(allObjects, objects.Items...)
 						}
-						allObjects = append(allObjects, objects.Items...)
 					}
 				}
+			} else if len(backup.Spec.ExcludeResourceTypes) != 0 {
+				isPersistentVolumeType := false
+				for _, rType := range backup.Spec.ExcludeResourceTypes {
+					if rType == "PersistentVolumeClaim" {
+						isPersistentVolumeType = true
```

Comment:
I think you should break here if you find PVC is part of excludeResourceType no need to loop through all the list. And infact we can eliminate the flag and append PV to the exclude here and break.

Reply:
Since I used IsPVCInExcludeResourceType(), I used flag